src/client/coll/mod.rs

```diff
@@ -0,0 +1,309 @@
+
+        let rp = match read_preference {
+            Some(rp) => rp,
+            None => db.read_preference.to_owned(),
+        };
+
+        let wc = match write_concern {
+            Some(wc) => wc,
+            None => db.write_concern.to_owned(),
+        };
+
+        Collection {
+            db: db,
+            namespace: format!("{}.{}", db.name, name),
+            read_preference: rp,
+            write_concern: wc,
+        }
+    }
+
+    /// Returns a unique operational request id.
+    pub fn get_req_id(&self) -> i32 {
+        self.db.client.get_req_id()
+    }
+
+    /// Extracts the collection name from the namespace.
+    pub fn name(&self) -> String {
+        match self.namespace.find(".") {
+            Some(idx) => self.namespace[idx+1..].to_owned(),
+            None => {
+                // '.' is inserted in Collection::new, so this should only panic due to user error.
+                let msg = format!("Invalid namespace specified: '{}'.", self.namespace);
+                panic!(msg);
+            }
+        }
+    }
+
+    /// Permanently deletes the collection from the database.
+    pub fn drop(&'a self) -> Result<(), String> {
+        self.db.drop_collection(&self.name()[..])
+    }
+
+    // Read Spec
+    /// Runs an aggregation framework pipeline.
+    pub fn aggregate(pipeline: &[bson::Document], options: AggregateOptions) -> Result<Vec<bson::Document>, String> {
+        unimplemented!()
+    }
+
+    /// Gets the number of documents matching the filter.
+    pub fn count(filter: bson::Document, options: CountOptions) -> Result<i64, String> {
+        unimplemented!()
+    }
+
+    /// Finds the distinct values for a specified field across a single collection.
+    pub fn distinct(field_name: &str, filter: bson::Document, options: DistinctOptions) -> Result<Vec<Bson>, String> {
+        unimplemented!()
+    }
+
+    /// Returns a list of documents within the collection that match the filter.
+    pub fn find(&self, filter: Option<bson::Document>, options: Option<FindOptions>)
+                -> Result<Vec<bson::Document>, String> {
+
+        let doc = match filter {
+            Some(bson) => bson,
+            None => bson::Document::new(),
+        };
+
+        let options = match options {
+            Some(opts) => opts,
+            None => FindOptions::new(),
+        };
+
+        let flags = OpQueryFlags {
+            tailable_cursor: options.cursor_type != CursorType::NonTailable,
+            slave_ok: false,
+            oplog_relay: options.op_log_replay,
+            no_cursor_timeout: options.no_cursor_timeout,
+            await_data: options.cursor_type == CursorType::TailableAwait,
+            exhaust: false,
+            partial: options.allow_partial_results,
+        };
+
+        let req = try!(Message::with_query(self.get_req_id(), flags, self.namespace.to_owned(),
+                                           options.skip as i32, options.limit, doc, options.projection));
```

Comment:
Is there any reason why `skip` isn't being stored as an int32 in all of the options structs?